cocotb/handle.py

```diff
@@ -316,6 +316,10 @@ def __setattr__(self, name, value):
         sub = self.__get_sub_handle_by_name(name)
         if sub is not None:
             sub.value = value
+            warnings.warn(
+                "Setting values on handles using the ``dut.handle = value`` syntax is deprecated."
+                "Instead use the ``handle.value = value`` or ``handle <= value`` syntax",
+                DeprecationWarning, stacklevel=2)
```

Comment:
I'd swap the order of these lines, so that if the user promotes warnings to errors, the assignment never happens.